graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/ClientApiGenerator.kt

```diff
@@ -130,30 +171,28 @@ class ClientApiGenerator(private val config: CodeGenConfig, private val document
                     """.trimMargin()
                 )
 
-            if (deprecatedDirective != null) {
-                methodBuilder.addAnnotation(java.lang.Deprecated::class.java)
-            }
-
-            // Build Javadoc, separate multiple blocks by empty line
-            val javaDoc = CodeBlock.builder()
-
-            if (inputValue.description != null) {
-                javaDoc.add("\$L", inputValue.description.content)
-            }
-            if (deprecationReason != null) {
-                if (!javaDoc.isEmpty) {
-                    javaDoc.add("\n\n")
-                }
-                javaDoc.add("@deprecated \$L", deprecationReason)
-            }
-
-            if (!javaDoc.isEmpty) {
-                methodBuilder.addJavadoc(javaDoc.build())
-            }
-
+            addDeprecationWarnings(deprecatedDirective, methodBuilder, inputValue, deprecationReason)
             builderClass.addMethod(methodBuilder.build())
                 .addField(findReturnType, ReservedKeywordSanitizer.sanitize(inputValue.name), Modifier.PRIVATE)
 
+            val inputValueType = inputValue.type
+            val typeForVariableDefinition = getVariableDefinitionType(inputValueType)
+            val referenceMethodBuilder = MethodSpec.methodBuilder(ReservedKeywordSanitizer.sanitize(inputValue.name) + "Reference")
+                .addParameter(stringType, "variableRef")
+                .returns(ClassName.get("", "Builder"))
+                .addModifiers(Modifier.PUBLIC)
+                .addCode(
+                    """
+                    |this.variableReferences.put("${ReservedKeywordSanitizer.sanitize(inputValue.name)}", variableRef);
+                    |this.variableDefinitions.add(graphql.language.VariableDefinition.newVariableDefinition(variableRef, $typeForVariableDefinition).build());
```

Comment:
extra nit: add import for `graphql.language.VariableDefinition` at the top

Reply:
you can't add imports explicitly with Javapoet